synapse/cortex.py

```diff
@@ -1206,6 +1222,74 @@ async def coreQueuePop(self, name, offs):
     async def coreQueueSize(self, name):
         return self.multiqueue.size(name)
 
+    @s_nexus.Pusher.onPushAuto('tag:meta:set')
+    async def setTagMeta(self, tagname, name, valu):
+
+        meta = self.taghive.get(tagname)
+        if meta is None:
+            meta = {}
+
+        meta[name] = valu
+        reqValidTagMeta(meta)
+
+        await self.taghive.set(tagname, meta)
+
+        # clear cached entries
+        if name == 'regex':
+            self.tagvalid.clear()
+
+    @s_nexus.Pusher.onPushAuto('tag:meta:del')
+    async def delTagMeta(self, tagname):
+        await self.taghive.pop(tagname)
+        self.tagvalid.clear()
+
+    @s_nexus.Pusher.onPushAuto('tag:meta:pop')
+    async def popTagMeta(self, tagname, name):
+
+        meta = self.taghive.get(tagname)
+        if meta is None:
+            return None
+
+        retn = meta.pop(name, None)
+        await self.taghive.set(name, meta)
+
+        if name == 'regex':
+            self.tagvalid.clear()
+
+        return retn
+
+    async def isTagValid(self, tagname):
```

Comment:
is there any reason that this needs to be async?

Reply:
I guess we could change it when it is inevitably needed via a remote call, just trying to avoid future backward-incompatible changes.